File: rust/INFO1_06_F.rs
use std::io::BufRead;

fn main() {
    let reader = std::io::stdin();

    let mut idt = Vec::new();

    reader.lock().lines().take(3).for_each(|line| {
        let str = line.unwrap();
        idt.push(str.parse::<i32>().unwrap());
    });

    let a = idt[0] != 0;
    let b = idt[1] != 0;
    let c = idt[2] != 0;

    println!("{}", !(a && b) || c);
}

File: haskell/INFO1_06_F.hs
main = do
  [a,b,c] <- take 3 <$> map read <$> lines <$> getContents :: IO [Int]
  let a' = a == 1
  let b' = b == 1
  let c' = c == 1
  print $ ( not (a' && b') ) || c'
